### ISSUE TITLE
$ONE Wallet Chrome Extension user experience + suggestions / improvements !!!

### ISSUE THREAD

**Issue by C10NU7** [harmony-one/onewallet#72]
_**Hello 👋, I create this issue after my experience with $ONE Wallet and to suggest a few improvements !**_

1. **Can we add icons for tokens ? These letters (token name) without an icon for tokens it looks awful!** 

![No icons!](https://user-images.githubusercontent.com/37934859/96769166-6c489000-13e7-11eb-9449-d741717db357.png)![Tokens icons](https://user-images.githubusercontent.com/37934859/96769180-6f438080-13e7-11eb-8cc8-655dd991c147.jpg)

**2. I tested recently [https://harmony.unifiprotocol.com/](url) and I noticed that after I connect with my wallet, _Unifi_ automatically reads the address of my wallet, and after that I can no longer change, add or use another address on _Unifi_ ! The only way Unifi can read another address ($ ONE account) is to uninstall the $ ONE wallet Chrome Extension, select that wallet address (account) I want to use this time and go to _Unifi_ website! This is strange and I honestly don't think it's the best solution to change my wallet addresses on a website or any dapps !!! Those guys from _Unifi_ said is something that should be implemented in wallet and they can't do nothing about it !!! So, we should be able to _switch between wallet addresses_ and the website to read them _automatically_ !**

![Main wallet](https://user-images.githubusercontent.com/37934859/96789439-d9ffb680-13fd-11eb-9b56-100654ea9b8d.png)
![second wallet](https://user-images.githubusercontent.com/37934859/96789471-eedc4a00-13fd-11eb-9b20-05d47a41d25a.png)

**3. _Disconnect_ $ONE Wallet from websites and dapps !!! We need this option added in wallet and to be able to _disconnect our wallet_ from _website_ and _dapps_ you don't intend anymore to use it or to have your wallet connected with them !**

**4. _Remove_ wallet address (account) !!! Option (remove button) to _remove_ any of my _addresses_ (accounts) which I don't want to use anymore !!! _Again_, you can do this only by _removing_ and _reinstalling_ the $ONE Wallet Chrome Extension, and this is not the right solution !!!** 

![Disconnect!](https://user-images.githubusercontent.com/37934859/96790909-349a1200-1400-11eb-965e-29bd726a285b.png)![Connect sites Remove wallet](https://user-images.githubusercontent.com/37934859/96790927-3a8ff300-1400-11eb-8025-e0b35417692f.png)

**5. _Reject/Confirm_ transactions without _password_ !!! It's frustrating and annoying as long as my wallet isn't _locked_, to be always asked to sign the transaction with the _password_ !!! I think isn't a security issue if my wallet is _unlocked_, to sign the transactions only with _CONFIRM_ instead of _PASSWORD_ !!! The _lock timer_ is enough in my opinion and is not a must have signing transactions with _password_ !!!**

**6. If is possible, I think a _notification_ (1 or !) on the _$ONE Wallet Chrome Extension Icon_ would be good !!! Please see the _notification_ on $ONE extension _icon_ (1) in the next picture !!!**

![Reject Confirm tx + notification @ ext icon](https://user-images.githubusercontent.com/37934859/96792139-6dd38180-1402-11eb-9ae3-495a9be049b2.png)

**7. After the _NEW_ $ONE Wallet Chrome Extension was promoted for use and the _older $ONE deprecated_, many users complained that it's annoying that they no longer have easy access to the "_Go to Harmony_" _dashboard_ in their wallet !!! So, maybe we can add this _button_ on first page/face of the wallet under deposit / send !!! It would be _very useful_ for many users !**

![GTH](https://user-images.githubusercontent.com/37934859/96792936-11716180-1404-11eb-9b60-e406a41e92a9.png)![Go To Harmony dashboard](https://user-images.githubusercontent.com/37934859/96793124-68773680-1404-11eb-82f7-1c12b6f2e670.png)

**8. Any chances to have this _$ONE Wallet_ Chrome Extension _phone_ compatible IOS/AND ??? In _2020_ a wallet is a _MUST_ have _compatible_ with the _phone_ !!! Many _dapps_ are using/implement _Wallet Connect_ with which you can connect easily your wallet on their website/dapps using your phone !!! _Unifi_ announced they are adding also the possibility to connect directly to their website from your phone using _Wallet Connect_ !!!**

![Wallet Connect web](https://user-images.githubusercontent.com/37934859/96794194-71690780-1406-11eb-86e4-57c240687841.png)![WalCon phone view](https://user-images.githubusercontent.com/37934859/96794200-7332cb00-1406-11eb-9d95-f411d9d49065.png)

**Comment by SebastianJ** [comment]
Just a bit of clarification for 2. / UniFi: They can add a button in their UI to disconnect from the current wallet, e.g. how it's done in Swoop: http://tools.harmony.one.s3.amazonaws.com/swoop-change-wallets.mov

I haven't looked at the wallet code in detail - but if there aren't events emitted for when you change the address or the active network then we should definitely add that so that the feature set is comparable to MetaMask (which makes it easy for dApps to detect when the wallet or the network changes).

**Comment by raptor1001** [comment]
@C10NU7 2. regarding https://harmony.unifiprotocol.com/, I think your feedback is really great. I am not sure who has developed this site but imo, there should be a logout button definitely. In the next version, you will be able to revoke the onewallet session like metamask.
![image](https://user-images.githubusercontent.com/54774811/96847951-a5452c80-1408-11eb-87b2-d960611ff4ee.png)

**Comment by C10NU7** [comment]
> @C10NU7 2. regarding https://harmony.unifiprotocol.com/, I think your feedback is really great. I am not sure who has developed this site but imo, there should be a logout button definitely. Anyway, In the next version, you will be able to revoke the onewallet session like metamask.
> ![image](https://user-images.githubusercontent.com/54774811/96847951-a5452c80-1408-11eb-87b2-d960611ff4ee.png)

@raptor1001 Thank you very much for replying ! Looking forward to the next version of the wallet :) . I will let also those guys from Unifi to know that they can as well add a "logout button" on their website !

**Comment by raptor1001** [comment]
@C10NU7 I don't understand the 8th point. The wallet connects part. What do you want to improve in the onewallet for that?

**Comment by C10NU7** [comment]
> @C10NU7 I don't understand the 8th point. The wallet connects part. What do you want to improve in the onewallet for that? Can you describe more details? Thanks

@raptor1001 The 8th point is strictly about phone interaction with the $ONE wallet ! At the moment we can't interact with this wallet using the phone! Some websites and dapps have integrated Wallet Connect and through Wallet Connect I am able to connect my trust wallet for example ! Well my question was, if we can make this $ONE Wallet to be also compatible with the phone as it is Metamask or Math wallet for example and be able to connect on dapps and website from phone !

Bottom line, $ONE Wallet compatible ios/android for phone usage !!! Thank you !

![photo_2020-10-22_17-29-48](https://user-images.githubusercontent.com/37934859/96886760-8f794b00-148c-11eb-8860-4317a6203c7e.jpg)![photo_2020-10-22_17-29-50](https://user-images.githubusercontent.com/37934859/96886775-92743b80-148c-11eb-8db0-d114717283d7.jpg)![photo_2020-10-22_17-31-31](https://user-images.githubusercontent.com/37934859/96886785-94d69580-148c-11eb-840e-534c45ab986c.jpg)

**Comment by raptor1001** [comment]
@C10NU7 oh so you are talking about the mobile ONEWallet?

**Comment by C10NU7** [comment]
> @C10NU7 oh so you are talking about the mobile ONEWallet?

@raptor1001 Exactly! $ONEWallet mobile 👍
